packages/ra-ui-materialui/src/field/ReferenceField.spec.tsx

```diff
@@ -612,6 +615,23 @@ describe('<ReferenceField />', () => {
             });
         });
     });
+    describe('Security', () => {
+        it('should render a link only when users have access to the requested action for the referenced resource', async () => {
+            render(<AccessControl />);
```

Comment:
This test is slow because of the default auth provider delay. Instead, build a custom auth provider in this test that uses a suspended Promise, pass this auth provider to the story, and resolve the Promise manually later in the test.